Algorithms.Tests/Numeric/FactorialTests.cs

```diff
@@ -38,5 +39,60 @@ public static void GetsFactorial(int input, long expected)
+        public static void GetsFactorial(string input, string expectedResult)
+        {
+            // Arrange
+            bool isInputConversionSuccessful = BigInteger.TryParse(input, out BigInteger inputBigInt);
+            bool isExpectedResultConversionSuccessful = BigInteger.TryParse(expectedResult, out BigInteger expectedResultBigInt);
+
+            if(isInputConversionSuccessful && isExpectedResultConversionSuccessful)
+            {
+                // Act
+                var result = Factorial.Calculate(inputBigInt);
+
+                // Assert
+                Assert.AreEqual(expectedResultBigInt, result);
+            }
+            else
+            {
+                Assert.AreEqual(true, false, "Cannot convert input parameters to BigInteger");
+            }
+        }
+
+        [Test]
+        [TestCase("-5")]
+        [TestCase("-100")]
+        public static void GetsFactorialExceptionForNonPositiveNumbers(string input)
```

Comment:
```suggestion
        public static void GetsFactorialExceptionForNegativeNumbers(string input)
```
0 is non-positive, but 0! == 1

Reply:
Issue resolved in new pull request